### PR TITLE
feat: New Home screen collections rail

### PR DESCRIPTION
**Pull request by olerichter00**
This PR resolves [CX-3412] 

### Description

This PR introduces the new Home screen collections rail (with a feature flag).

| Before | After |
| --- | --- |
| Simulator Screen Shot - iPhone 14 Pro - 2023-02-22 at 15 20 32| Simulator Screen Shot - iPhone 14 Pro - 2023-02-22 at 15 18 52|

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- New Home screen collections rail - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[CX-3412]: https://artsyproduct.atlassian.net/browse/CX-3412?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Components/FiveUpImageLayout.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      </Flex>
+    </Flex>
+  )
+}
+
+const backfillImageURLs = (imageURLs: string[], arrayLength: number) => {
+  return Array.from({ length: arrayLength }, () => null).reduce((acc: string[], _, i) => {
+    return [...acc, imageURLs[i] || acc[i - 1]]
+  }, [])
+}
```
[the remark this answers is not in the document]
Comment (lordkiz) [reply src/app/Components/FiveUpImageLayout.tsx:65]: ***Thought***: I wonder whether you don't want to vary the next image by picking from an index distant from the last. That way two images of the same type do not lie side by side.

**Review by lordkiz — APPROVED** [review]
Looks great

File: src/app/Components/FiveUpImageLayout.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (lordkiz) [reply src/app/Components/FiveUpImageLayout.tsx]: Better. But I think that function will fail(or populate with `"undefined"s` if the difference between arrayLength and imageURLs.length is greater than one.

[the remark this answers is not in the document]
Comment (olerichter00) [reply src/app/Components/FiveUpImageLayout.tsx]: I tested it and it works fine because of the [% operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Remainder) (`imageURLs[i % imageURLs.length]`)

[the remark this answers is not in the document]
Comment (lordkiz) [reply src/app/Components/FiveUpImageLayout.tsx]: I mean, for eg if imageURLs has a length of 2 and you need arrayLength 5. By the time `i` is 3, you would run `imageURLs[3]` which would return undefined.

[the remark this answers is not in the document]
Comment (olerichter00) [reply src/app/Components/FiveUpImageLayout.tsx]: It's not possible to run into an index that is larger than `imageURLs - 1`.

`imageURLs[3 % 2] == imageURLs[1]` because `3 % 2 == 1`.

[the remark this answers is not in the document]
Comment (lordkiz) [reply src/app/Components/FiveUpImageLayout.tsx]: @olerichter00 Oh sorry. Somehow I was reading this as `[i % arrayLength]` instead of `[i % imageURLs.length]`. 😂

**Review by lordkiz — DISMISSED** [review]

File: src/app/Components/FiveUpImageLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          source={{ uri: artworkImageURLs[4] }}
+          resizeMode="cover"
+        />
+      </Flex>
+    </Flex>
+  )
+}
+
+const backfillImageURLs = (imageURLs: string[], arrayLength: number) => {
+  return Array.from({ length: arrayLength }, (_, i) => imageURLs[i % imageURLs.length])
```
Comment (MounirDhahri) [inline src/app/Components/FiveUpImageLayout.tsx:66]: **praise:** nice use for the modulo!

File: src/app/Scenes/Home/Components/CollectionsRail.tsx [hunk 3]
```diff
@@ -48,45 +45,44 @@ const CollectionsRail: React.FC<Props & RailScrollProps> = (props) => {
...
           // still be cautious to avoid crashes if this assumption is broken.
           const artworkImageURLs = extractNodes(
-            result.artworksConnection,
+            item.artworksConnection,
             (artwork) => artwork.image?.url!
           )
 
           return (
-            <CardRailCard
+            <TouchableWithoutFeedback
```
Comment (MounirDhahri) [inline src/app/Scenes/Home/Components/CollectionsRail.tsx:57]: **nitpick:** is this change intentional? I was thinking that we might be able to use haptic touchable here to provide feedback about his

**Review by MounirDhahri — APPROVED** [review]
Nice - only one minor non-launch blocking comment.

File: src/app/Scenes/Home/Components/CollectionsRail.tsx (hunk 3 above)
Comment (olerichter00) [reply src/app/Scenes/Home/Components/CollectionsRail.tsx]: Yes, it's intentional. `CardRailCard` has a border that we don't need anymore.

I'll change it to use `TouchableHighlight` so we have a touch opacity effect similar to other rails 🙏
